backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -348,17 +358,50 @@ public Response create(@PathParam("vaultId") UUID vaultId, @Valid VaultDto vault
 		}
 	}
 
+	@PATCH
+	@Path("/{vaultId}")
+	@RolesAllowed("admin")
+	@Transactional
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	@Operation(summary = "Updates a vault",
+			description = "Changes the vault description, its name or its archive status.")
+	@APIResponse(responseCode = "200", description = "the vault has been updated")
+	@APIResponse(responseCode = "403", description = "requesting user does not have admin role.")
+	public VaultDto update(@PathParam("vaultId") UUID vaultId, VaultUpdateDto update) {
+		Vault vault = Vault.<Vault>findByIdOptional(vaultId).orElseThrow(NotFoundException::new);
+		if (update.name != null) {
+			vault.name = update.name;
+		}
+
+		if (update.description != null) {
+			vault.description = update.description;
+		}
+
+		if (update.archived != null) {
+			if (!("false".equalsIgnoreCase(update.archived) || "true".equalsIgnoreCase(update.archived))) {
+				throw new BadRequestException("Allowed values for field \"archived\" are [true, false]");
+			}
+			vault.archived = Boolean.valueOf(update.archived);
+		}
```

Comment:
I dislike the `if x != null` part.

The request body shall represent the whole truth. In ReST APIs I don't expect the backend to reinterpret it. Updating must include removing one or more of the resource's properties (maybe I want to delete a vault's description).

If the frontend likes to keep things unchanged, it should send them unchanged, not exclude them from a DTO.

Reply:
That is part of the PATCH contract, at least to the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH):
>A PATCH request is considered a set of instructions on how to modify a resource. Contrast this with [PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT); which is a complete representation of a resource.

Reply:
I don't think that by "complete" the spec means the same as non-null in Java. Imagine a resource with the initial state `{"a": 1, "b": null, "c": "hello"}`. Now let's patch it using `{"b": "world", "c:" null}`. I would expect the resource to be `{"a": 1, "b": "world", "c": null}` now.

However, the current approach doesn't allow us to distinguish between "I want a field to be null" and "I don't want to modify a field", partly due to how Java/Jackson work.

If we instead contractually specify that we always send the "full" json representation in the request body, we don't have to deal with "missing" entries. If the sender doesn't want a value to change, he must not change it (but still send it). Furthermore, this allows better protection against race conditions: If the sender sends `{"a": 1, ...}` but the resource has been changed to `a=2` on the server, we can easily implement server-side checks and respond with 428. If however, `a` was omitted, the server would update the resource, leaving the client under the false impression that `a` would still be `1`.

Reply:
Obsolete due to https://github.com/cryptomator/hub/pull/202#discussion_r1231961355